### PR TITLE
feat(SCFD-2358): Remove 50G upload limit on python client

### PR DESCRIPTION
**Pull request by benflexcompute**
Supposedly the upload limit is now 5T.https://flow360.atlassian.net/browse/SCFD-2487

### CODE REVIEW

File: flow360/cloud/s3_utils.py [hunk 1]
```diff
@@ -281,7 +300,7 @@ def upload_file(
                 Filename=file_name,
                 Key=token.get_s3_key(),
                 Callback=progress_callback,
-                Config=_s3_config,
+                Config=_get_dynamic_upload_config(float(os.path.getsize(file_name))),
```
Comment (lei-flex) [inline flow360/cloud/s3_utils.py:303]: why change to float?

**Review by lei-flex — CHANGES_REQUESTED** [review]

File: flow360/cloud/s3_utils.py (hunk 1 above)
Comment (benflexcompute) [reply flow360/cloud/s3_utils.py]: Removed float

File: flow360/cloud/s3_utils.py [hunk 2]
```diff
@@ -38,12 +39,31 @@ def __call__(self, bytes_chunk_transferred):
...
+def _get_dynamic_upload_config(file_size) -> TransferConfig:
+    # Predefined chunk sizes
+
+    def select_chunk_size(file_size, max_parts=10000):
+        """
+        Select an appropriate chunk size for a given file size based on predefined chunk sizes.
+
+        :param file_size: Size of the file to be uploaded in bytes.
+        :param chunk_sizes: List of chunk sizes in bytes to consider.
+        :param max_parts: Maximum number of parts allowed for multipart upload.
```
Comment (benflexcompute) [inline flow360/cloud/s3_utils.py:51]: remove function and inline. Link doc reference

**Review by lei-flex — APPROVED** [review]
ship it

**Pull request merged**